wandb/wandb_agent.py

```diff
@@ -241,6 +241,22 @@ def run(self):  # noqa: C901
                             self._running = False
                             break
                     logger.info("Cleaning up finished run: %s", run_id)
+
+                    # wandb.teardown() was added with wandb service and is a hammer to make
```

Comment:
i see we went with the large hammer approach

Reply:
I really considered the more delicate approach with inform_finish, i didnt prototype it but i ended up being consider about agent runs that spawned other runs.... I think we could handle that when we have the run hierarchy  stuff, but it would be tricky and more risky to do that approach right now.

Reply:
yeah this one works... just adds some teardown and spinup time, but i don't think it is that big perf issue

Reply:
Sorry about merging this. I saw green and impulse clicked.. :(